Update AOD handling of ambiguous MFT and Fwd tracks

rpezzi opened the pull request:
* Adds TrackTime and TrackTimeResolution for forward track table
* Ambiguous MFT and forward tracks assigned to BC Slice
* Add comments to Fwd tracks tables element definitions

The PR uses same column definitions for ambiguous as #6970 (and will result in merge conflicts) with it.

jgrosseo commented on Framework/Core/include/Framework/AnalysisDataModel.h:
```diff
@@ -332,17 +332,18 @@ using FullTrack = FullTracks::iterator;
+DECLARE_SOA_INDEX_COLUMN(Collision, collision); //! Collision to which this track belongs
+DECLARE_SOA_INDEX_COLUMN(BC, bc);               //! BC to which this track belongs
+// TODO change to ForwardTrackTypeEnum when enums are supported
+DECLARE_SOA_COLUMN(TrackType, trackType, uint8_t);             //! Type of track. See enum TrackTypeEnum
```
Shouldn't this be ForwardTrackTypeEnum?

rpezzi replied:
fixed

jgrosseo commented:
Thanks! Can you use also self index (DECLARE_SOA_SELF_INDEX_COLUMN_FULL) for MatchMCHTrackID and a index to the MFT standalone track table for MatchMFTTrackID? Or am I missing something here?

jgrosseo commented on Framework/Core/include/Framework/AnalysisDataModel.h:
```diff
@@ -332,26 +332,29 @@ using FullTrack = FullTracks::iterator;
+DECLARE_SOA_COLUMN(Chi2MatchMCHMFT, chi2MatchMCHMFT, float);                              //! MCH-MFT Match Chi2 for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchScoreMCHMFT, matchScoreMCHMFT, float);                            //! MCH-MFT Machine Learning Matching Score for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchMFTTrackID, matchMFTTrackID, int);                                //! ID of matching MFT track for GlobalMuonTracks
+DECLARE_SOA_SELF_INDEX_COLUMN_FULL(MatchMCHTrack, matchMCHTrack, int, "MatchMCHTrackID"); //! ID of matching MCH track for GlobalMuonTracks
```
In the file this should be the column `fIndexFwdTracks_MatchMCHTrackID`
You can probably reduce MatchMCHTrackID. At least ID can be removed.

jgrosseo commented on Framework/Core/include/Framework/AnalysisDataModel.h:
```diff
@@ -332,26 +332,29 @@ using FullTrack = FullTracks::iterator;
+DECLARE_SOA_COLUMN(Chi2MatchMCHMID, chi2MatchMCHMID, float);                              //! MCH-MID Match Chi2 for MUONStandalone tracks
+DECLARE_SOA_COLUMN(Chi2MatchMCHMFT, chi2MatchMCHMFT, float);                              //! MCH-MFT Match Chi2 for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchScoreMCHMFT, matchScoreMCHMFT, float);                            //! MCH-MFT Machine Learning Matching Score for GlobalMuonTracks
+DECLARE_SOA_COLUMN(MatchMFTTrackID, matchMFTTrackID, int);                                //! ID of matching MFT track for GlobalMuonTracks
```
Can this be an index column to MFTStandalone?

rpezzi commented on Framework/Core/include/Framework/AnalysisDataModel.h:
```diff
@@ -438,17 +437,22 @@ DECLARE_SOA_EXTENDED_TABLE(MFTTracks, StoredMFTTracks, "MFTTRACK", //!
+namespace fwdtrack // Index to MFTtrack column must be defined after table definition. Should a SELF_INDEX_COLUMN be used here?
+{
+DECLARE_SOA_INDEX_COLUMN(MFTTrack, matchMFTTrack); //! ID of matching MFT track for GlobalMuonTracks
+}
```
@jgrosseo , notice here that in order to define the index to MFT track column, the corresponding mfttrack table must be already defined. This breaks the column definitions in two parts.

jgrosseo replied:
The index is right and also your observation. You can remove "Should a SELF_INDEX_COLUMN be used here?"

rpezzi commented:
@jgrosseo ,
new commit with change discussed this morning:
* BCId removed from fwdtrack table
* Added ambiguousfwdtracks table with SLICE_INDEX_COLUMN colum for BCs and _ARRAY_INDEX_COLUMN for Collisions.

I understand track-times will be added to central barrel tracks and we would like to have it in the fwd track table as well.

jgrosseo commented:
Thanks for the update. I have merged the PR which keeps only the BC for the ambiguous table, you can merge it with this one, and you need to align only the AMBIGUOUSFWDTR (I already did the mft one). I also have a comment already that in the future we will add the collision indices.

jgrosseo commented:
The track time update is here https://github.com/AliceO2Group/AliceO2/pull/6986
However, we have to wait for adjusting the Run 3 converter and new data before merging. The same will be for this PR.

jgrosseo commented:
@rpezzi Can you rebase and align the ambiguous table with the ones in the data model now (BC only)? We also need to adjust the RUN2 converter in AliPhysics

rpezzi committed:
Update AOD table for fwd tracks creates table for ambiguous fwdtracks

- Comment FwdTracks/FwdTracksCov columns definitions
- Adapt AOD producer workflow

rpezzi commented:
> @rpezzi Can you rebase and align the ambiguous table with the ones in the data model now (BC only)? We also need to adjust the RUN2 converter in AliPhysics

Hi @jgrosseo , this has been updated. I have included tracktimes in the fwdTrack table, but I see that #6986 has not been merged yet. This info will be readly available after PR #6966 is merged.

jgrosseo commented:
> > @rpezzi Can you rebase and align the ambiguous table with the ones in the data model now (BC only)? We also need to adjust the RUN2 converter in AliPhysics
> 
> Hi @jgrosseo , this has been updated. I have included tracktimes in the fwdTrack table, but I see that #6986 has not been merged yet. This info will be readly available after PR #6966 is merged.

Thanks. It will be available for RUN3 AODs but not for RUN2 converted files. So the fwdtracks table of RUN2 converted data will be unreadable I think. This has to be adjusted in AliPhysics.

rpezzi committed:
Symmetric naming of MFT and MCH track-match IDs

rpezzi commented:
This PR should be merged in sync with https://github.com/AliceO2Group/O2Physics/pull/84

rpezzi committed:
Fix MCHTrack self index column definition
